### ISSUE TITLE
How can I limit one of the submodels of an ensemble to run on CPU only?

### ISSUE THREAD

**Issue by Alwahsh** [triton-inference-server/model_analyzer#621]
If I want to run a normal model on the CPU only, I can do this in the YAML configuration file:
```
profile_models: 
  model_name:
    cpu_only: false
```

How can I achieve the same effect for one of the submodels in an ensemble model? I don't want all submodels to run on the CPU only but I want some of them to run on the CPU and others on the GPU.

Thanks.

**Comment by tgerdesnv** [comment]
@Alwahsh, that is a very good question. @nv-braf would be the person that knows for sure but he probably won't be available to reply until next week. I do see an internal ticket that implies this might not be supported yet.

May I ask are you running out of the 23.02 container? Or are you pulling from github?

**Comment by Alwahsh** [comment]
Thank you! 
I'm running out of 23.02 container but I had to change my configuration to use 23.01 version of Triton since the container of 23.02 doesn't seem to be out yet.
And I did manage to get the ensemble testing working. Thanks for providing it :pray:

**Comment by nv-braf** [comment]
This is currently not supported.

**Comment by dmastapkovich** [comment]
> This is currently not supported.

Is `cpu_only` support planned? If so, when is it planned?

**Comment by nv-braf** [comment]
Yes, this will be supported as of 23.05 release, which should be by the end of the month (or you can grab it off of main). The new option is called `--cpu-only-composing-models`

**Comment by dmastapkovich** [comment]
@nv-braf There is no support for `--cpu-only-composing-models` in `model_analyzer` version `1.28`?

**Comment by tgerdesnv** [comment]
Yes, there should be support for `--cpu-only-composing-models` in `1.28`, which is the version in the 23.05 release.  Is it not working for you?

**Comment by dmastapkovich** [comment]
@tgerdesnv 
I use `model_analyzer` locally and install it via `pip`:
```
pip install triton-model-analyzer
```

there is no functionality in the archive with `--cpu-only-composing-models`
```
[Model Analyzer] cpu_only_composing_models is not a valid yaml argument. Please be sure to use underscores and check spelling.
[Model Analyzer] Model Analyzer encountered an error: The Yaml configuration file is incorrect. Please check the logged errors for the specific options that are causing the issue.
```

But it works for me if you install through pip like this 
```
pip install git+https://github.com/triton-inference-server/model_analyzer.git
```

**Comment by tgerdesnv** [comment]
You are correct. Unfortunately it appears that the feature didn't quite make it into 23.05.  As you observed, you can install off of main and it will work. Is that sufficient for you for now? It will definitely be in 23.06.

**Comment by dmastapkovich** [comment]
So-so, but I will wait. Thank you

**Comment by dyastremsky** [comment]
This feature was released starting in 23.06. See documentation [here](https://github.com/triton-inference-server/model_analyzer/blob/fe049871b5bf3ab244152e21708c67aade17f4dc/docs/config.md?plain=1#L96).
